spec/features/user_sign_ups_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+feature 'user signs up' do
+  scenario 'with valid info' do
+    visit root_path
+    click_link 'Sign Up'
+    fill_in 'Email', with: 'user@example.com'
+    fill_in 'Password', with: 'supersecret', match: :prefer_exact
+    fill_in 'Password confirmation', with: 'supersecret'
+    click_button 'Sign up'
+    expect(page).to_not have_link 'Sign Up'
+    expect(page).to have_link 'Sign Out'
```

Comment:
In tests, I try to separate the Data setup, Actions and then Expectations. I think it makes it easier to follow/change.

So here, I'd do it like this.
```Ruby
  visit root_path
  click_link 'Sign Up'
  fill_in 'Email', with: 'user@example.com'
  fill_in 'Password', with: 'supersecret', match: :prefer_exact
  fill_in 'Password confirmation', with: 'supersecret'
  click_button 'Sign up'

  expect(page).to_not have_link 'Sign Up'
  expect(page).to have_link 'Sign Out'
```